vscode/src/test/suite/testutils.ts

```diff
@@ -192,3 +192,278 @@ export async function dumpJava(): Promise<void> {
+        if (!keys.has(key)) {
+            console.log(`Found invalid localisation key in file:'${filePath.replace(".js", ".ts")}'.Here is the expression used in file with invalid key '${context}'`);
+            result = false;
+            continue;
+        }
+        if (placeholders != undefined) {
+            if (!placeholderMatch(validKeyValues[key], placeholders)) {
+                console.log(placeholders);
+                result = false;
+                console.log(`Wrong placeholder map for a localisation key  in file:'${filePath.replace(".js", ".ts")}'.Here is the expression used in file with wrong placeholder map '${context}'. Here is the bundle value '${validKeyValues[key]}'`);
+            }
+        } else {
+            if (!matchTemplate("", validKeyValues[key])) {
+                result = false;
+                console.log(`Placeholder map not provided for a localisation key in file:'${filePath.replace(".js", ".ts")}'.Here is the expression used in file without the placeholder '${context}'. Here is the bundle value '${validKeyValues[key]}'`);
+            }
+        }
+    }
+    return result;
+}
+
+/**
+ * 
+ * @param value string to be checked
+ * @returns value of the form %key% where key is some string containing alphanumeric characters 
+ */
+function isLocalizedVal(value: string): boolean {
+    return (value[0] === value[value.length - 1]) && value[0] === '%';
+}
+
+
+/**
+ * 
+ * @param str localised value of the form %key%
+ * @returns key 
+ */
+function getlocalizingKey(str: string): string {
+    const length = str.length;
+    return str.substring(1, length - 1);
+}
+
+/**
+ * 
+ * @param obj 
+ * @param localisableFields Array or Scalar fields which are to be tested for localisation 
+ * @param id 
+ * @param category 
+ * @param validKeys Keys present in the package.nls.json
+ * @returns Whether the object given has the required fields localised by some valid key 
+ */
+function isLocalizedObj(obj: any, localisableFields: any, id: string, category: string, validKeys: Set<string>): boolean {
+    let localized: boolean = true;
+    let fieldVals: any;
+    let localizingKey: string;
+    for (const field of localisableFields) {
+        fieldVals = obj[field];
+        if (fieldVals === undefined) continue;
+        if (!Array.isArray(fieldVals)) fieldVals = [fieldVals];
+        for (const fieldVal of fieldVals) {
+            localizingKey = getlocalizingKey(fieldVal);
+            if (!isLocalizedVal(fieldVal)) {
+                console.log(`${category} object with id ${id} has a unlocalized field field:'${field}'`);
+                localized = false;
+            } else if (!validKeys.has(localizingKey)) {
+                console.log(`${category} object of id '${id}' has a invalid  localizing key for the field:'${field}' key:'${localizingKey}'`);
+                localized = false;
+            }
```

Comment:
To avoid an array index out of bounds err.
```suggestion
            if (!isLocalizedVal(fieldVal)) {
                console.log(`${category} object with id ${id} has a unlocalized field field:'${field}'`);
                localized = false;
            } else {
                localizingKey = getlocalizingKey(fieldVal);
                if (!validKeys.has(localizingKey)) {
                    console.log(`${category} object of id '${id}' has a invalid  localizing key for the field:'${field}' key:'${localizingKey}'`);
                    localized = false;
                }
            }
```

Reply:
Sorry, I wrote arr index out of bounds instead of a possible string index err in `getlocalizingKey(fieldVal)`.
However, I just realised that the js `string.substring()` does not throw an err, but instead will return an unexpected output when the `endIndex < beginIndex`. 
So, the change in order of the method invocations - first `isLocalizedVal`, then `getlocalizingKey` will only have a minor performance improvement.

Reply:
Then I guess we are okay not changing it right ?